client/client.go

```diff
@@ -316,52 +309,56 @@ type Config struct {
 	NoVerify   bool
 }
 
-func loadServerTrust(conf Config) (*x509.CertPool, error) {
-	if conf.ServerCert == "" {
-		// Returning nil will make tls.Config.RootCAs nil, which causes
-		// tls module to use system certs.
-		return nil, nil
-	}
+type systemCertPoolGetter interface {
+	GetSystemCertPool() (*x509.CertPool, error)
+}
+
+type systemCertPool struct{}
+
+func (systemCertPool) GetSystemCertPool() (*x509.CertPool, error) {
+	return x509.SystemCertPool()
+}
+
+func loadServerTrust(conf *Config) *x509.CertPool {
+	return loadServerTrustImpl(conf, systemCertPool{})
+}
 
-	syscerts, err := x509.SystemCertPool()
+func loadServerTrustImpl(conf *Config, scp systemCertPoolGetter) *x509.CertPool {
+	syscerts, err := scp.GetSystemCertPool()
 	if err != nil {
-		return nil, err
+		log.Warnf("Error when loading system certificates: %s", err.Error())
 	}
 
 	// Read certificate file.
 	servcert, err := ioutil.ReadFile(conf.ServerCert)
 	if err != nil {
-		log.Errorf("%s is inaccessible: %s", conf.ServerCert, err.Error())
-		return nil, err
-	}
-
-	if len(servcert) == 0 {
-		log.Errorf("Both %s and the system certificate pool are empty.",
-			conf.ServerCert)
-		return nil, errors.New("server certificate is empty")
-	}
-
-	block, _ := pem.Decode([]byte(servcert))
-	if block != nil {
-		cert, err := x509.ParseCertificate(block.Bytes)
-		if err == nil {
-			log.Infof("API Gateway certificate (in PEM format): \n%s", string(servcert))
-			log.Infof("Issuer: %s, Valid from: %s, Valid to: %s",
-				cert.Issuer.Organization, cert.NotBefore, cert.NotAfter)
-		}
+		// Ignore server certificate error  (See: MEN-2378)
+		log.Warnf(errMissingServerCertF, err.Error())
 	}
 
 	if syscerts == nil {
```

Comment:
I guess this is not necessary anymore then?

Reply:
It is necessary, because this initializes the list to an empty one if `GetSystemCertPool()` returned `nil`. I should add a test case for that, actually...